syntax = "proto3";

package tensorflow;
// option cc_enable_arenas = true;

message AllocationDescription {
  // Total number of bytes requested
  int64 requested_bytes = 1;

  // Total number of bytes allocated if known
  int64 allocated_bytes = 2;

  // Name of the allocator used
  string allocator_name = 3;

  // Identifier of the allocated buffer if known
  int64 allocation_id = 4;

  // Set if this tensor only has one remaining reference
  bool has_single_reference = 5;
};
